### PR TITLE
Logging in sync

### PR DESCRIPTION
**Pull request by pawelgocek**
This feature adds sync progress logging through SLF4J.

### CODE REVIEW

**Comment by bmista** [comment]
I guess we should do the same for Jersey client by introducing bridge between JUL and SLF4J.

File: build.gradle [hunk 1]
```diff
@@ -31,11 +31,13 @@ dependencies {
     compile group: "com.fasterxml.jackson.core", name: "jackson-databind", version: "2.5.0"
     compile group: "joda-time", name: "joda-time", version: "2.3"
     compile group: "com.fasterxml.jackson.datatype", name: "jackson-datatype-joda", version: "2.0.4"
+    compile 'org.slf4j:slf4j-api:1.7.12'
```
Comment (michal-mally) [inline build.gradle]: Lets keep the convention. If we use map syntax (group:, name:, etc) lets keep it consistent.

File: build.gradle [hunk 2]
```diff
@@ -31,11 +31,13 @@ dependencies {
...
 
     testCompile group: 'junit', name: 'junit', version: '4.11'
     testCompile group: "org.spockframework", name: "spock-core", version: "1.0-groovy-2.4"
     testCompile group: "cglib", name: "cglib-nodep", version: "3.1"
     testCompile group: "org.objenesis", name: "objenesis-parent", version: "1.2"
+    testCompile 'ch.qos.logback:logback-classic:1.1.3'
```
Comment (michal-mally) [inline build.gradle]: Same as above.

File: src/main/java/com/getbase/sync/SyncService.java [hunk 3]
```diff
@@ -29,8 +34,20 @@ public Session start(String deviceUUID) {
                 buildHeaders(deviceUUID),
                 null);
 
-        if (response.getHttpStatus() == 204) return null;
-        return SessionDeserializer.deserialize(JsonDeserializer.deserializeRaw(response.getBody()));
+        if (response.getHttpStatus() == 204) {
```
Comment (michal-mally) [inline src/main/java/com/getbase/sync/SyncService.java]: Lets define what 204 (as const or use existing mapping from classpath if any exists)

File: src/main/java/com/getbase/sync/SyncService.java [hunk 4]
```diff
@@ -29,8 +34,20 @@ public Session start(String deviceUUID) {
...
+            log.info("Received 204 from sync start. Nothing to fetch.");
+            return null;
+        }
+
+        final Session session = SessionDeserializer.deserialize(JsonDeserializer.deserializeRaw(response.getBody()));
+
+        if (session != null) {
+            log.info("Started sync process with pending queues: {}", session.getQueues());
+        } else {
+            log.warn("Sync process will not be started due to missing session data: {}", response);
```
Comment (michal-mally) [inline src/main/java/com/getbase/sync/SyncService.java:50]: Isn't it an exceptional conditional that shall be handled by throwing exception?

File: src/main/java/com/getbase/sync/SyncService.java [hunk 5]
```diff
@@ -49,6 +66,7 @@ public Session start(String deviceUUID) {
 
         // nothing new to synchronize
         if (response.getHttpStatus() == 204) {
```
Comment (michal-mally) [inline src/main/java/com/getbase/sync/SyncService.java]: This condition might be extracted as method like `isSyncFinished()` not to duplicate it and for documentation purposes.

File: src/main/java/com/getbase/sync/SyncService.java [hunk 6]
```diff
@@ -81,7 +116,16 @@ public boolean ack(String deviceUUID, List<String> ackKeys) {
...
                 serialized);
 
-        return response.getHttpStatus() == 202;
+        boolean acked = response.getHttpStatus() == 202;
+
+        if (!acked) {
+            log.warn("Ack request was not accepted. Http status: {}. Response body: {}", response.getHttpStatus(),
+                    response.getBody());
+        } else {
+            log.debug("Items acknowledged.");
```
Comment (michal-mally) [inline src/main/java/com/getbase/sync/SyncService.java]: Don't we want to know what items where acknowledged or how many?

**Comment by michal-mally** [comment]
@pawelgocek please review my comments.

File: src/main/java/com/getbase/sync/SyncService.java (hunk 4 above)
Comment (pawelgocek) [inline src/main/java/com/getbase/sync/SyncService.java:50]: Agree but let's do it on separate issue.

**Pull request merged**